### PR TITLE
refactor: split wallet component into different components

### LINKED ISSUE #1288: Refactor Wallet component from `src/components/cards/Wallet.tsx`

### PR DESCRIPTION
**Pull request by mhenriette**
# Submit a pull request

- [x] This is not a duplicate of an existing pull request.
- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [x] The code style guideline have been followed.
- [x] Documentation has been updated to reflect your changes.
- [x] Tests have been added or updated to reflect your changes.
- [x] All tests pass.

---

## Pull Request Details

This PR refactors the existing Wallet component into several distinct, smaller components. The refactoring aims to improve code organization, maintainability, and reusability. 

## Breaking Changes

none

## Issues Fixed

Closes #1288 

## Other Relevant Information
none

### CODE REVIEW

File: src/components/cards/wallet/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <div className="relative mb-7" data-testid={testId}>
+      <div className="relative lg:flex md:flex sm:flex rounded-3.5xl">
+        {showEditModal && <EditAddress show={showEditModal} onClose={onClose} wallet={wallet} />}
+        <Payout wallet={wallet} show={showPayoutModal} onClose={onClose} />
+        <Overview wallet={wallet} />
+        <CashoutAddress wallet={wallet} setShowEditModal={setShowEditModal} disabled={disabled} setShowPayoutModal={setShowPayoutModal} />
+      </div>
+      <WalletHint wallet={wallet} />
```
Comment (serapieTuyishime) [inline src/components/cards/wallet/index.tsx:45]:
```suggestion
      <Hints wallet={wallet} />
```

**Review by serapieTuyishime — COMMENTED** [review]
I suggest to put the files in a `_partials` folder inside the `src/components/cards/wallet/`

File: src/components/cards/wallet/CashoutAddress.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            <p className="leading-5 text-sm flex gap-x-2 gap-y-1 flex-wrap font-mono font-normal">
+              {address.map((part, index) => (
+                <span key={`address-${index}`} className="mr-2">
+                  {part}
+                </span>
+              ))}
+            </p>
+          ) : (
+            <p>{wallet.description}</p>
+          )}
```
Comment (Jonath-z) [inline src/components/cards/wallet/CashoutAddress.tsx:72]: Can you make this a separate component?

File: src/components/cards/wallet/WalletHint.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Wallet hint component
+ *
+ * @returns {ReactElement}
+ */
+
+export default function WalletHint({ wallet }: WalletHintProps): ReactElement {
+  const { t } = useTranslation();
+  return (
+    <>
```
Comment (Jonath-z) [inline src/components/cards/wallet/WalletHint.tsx:20]: There is not need of empty fragment

Comment (mhenriette) [reply src/components/cards/wallet/WalletHint.tsx:20]: The map is returning multiple elements, we need these. fragments to return a single jsx

**Review by Jonath-z — APPROVED** [review]

**Review by serapieTuyishime — APPROVED** [review]

**Review by musayann — APPROVED** [review]

**Pull request merged**